### ISSUE TITLE
Change Incoming Scan notification

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#4240]
**Problem:**
 - Scanning messages clog the message section.
 - In busy systems it may take longer to identify which ship is doing the scanning than it takes for them to do the scan, making it very hard to evade the scan.

**Describe the solution you'd like**
 - Messages regarding scanning are no longer displayed in the message section.
 - When the player is being scanned, the scanning reticle appears around their ship indicating that a scan is in progress, accompanied by some kind of directional icon indicating the direction the scan is coming from. (This would not indicate the timing of the scan or how long it will take to complete it, since it doesn't seem reasonable that the player would be seeing the feedback from the other ship's scanners)
 - A color change, or pulse of some kind to indicate the scan completed.

**Comment by comnom** [comment]
>Scanning messages clog the message section.

I agree with this, but don't care for the idea of a special graphic in the main panel for scanning.

Better I think would be something in the radar to indicate a ship is scanning a target, not just the player (handholding again).

That would be consistent with how missiles are handled (in the radar), and also special target ships and other "hostile" information.

**Comment by Zitchas** [comment]
A) How often do NPCs scan other NPCs? The player doesn't really see this now, and this is very much irrelevant information. Changing things so that the player sees all scans that are occurring would be a change in what information is being relayed to the player, whereas the purpose of this request is to simply change the *way* this information is presented, not change the *amount* of information provided

B) In terms of gameplay impact, my desired impact would be to change dodging scans from simply being a binary thing (they do/don't happen) to something with enough feedback that a player running smuggling or other illegal activities becomes a kind of mini-game where they have to weave around ships trying to scan them. This sort of activity would be enhanced by having on-screen indicators showing which ships are doing the scan.

C) while it might be hand-holding, I would also consider it smart sensor design. Rather like how present day fighter jets tend to use "dark cockpit" design where the automated systems are designed to filter out all the irrelevant information so that the pilot is only informed about things that actually matter to them. For instance, modern fighters doesn't warn about every missile in the sky, just the ones that are locked onto them.

D) I'm not fond of the idea of putting scan indications onto the mini-map as it is already rather dense, and I'm concerned that adding some kind of scanning icon onto it would make it hard to see other details (like missiles).

That being said, maybe we could put an indicator around the minimap so that there is an indicator that appears when a scan starts indicating what direction it is coming from? That wouldn't clutter the minimap itself, and keep the info in that area.

**Comment by comnom** [comment]
>How often do NPCs scan other NPCs? The player doesn't really see this now, and this is very much irrelevant information.

That's fair, but the idea was actually to obfuscate the information a bit to retain some of the challenge. With the original idea, we get not only the exact direction of what's scanning you, but also the exact timing, which really doesn't make for a very interesting "mini-game" in my book at least.

>while it might be hand-holding, I would also consider it smart sensor design.

I agree with the reasoning, but again, where is the actual challenge if we feed all the information to the player? I'd just make sure my ship was faster than most, and easily evade it as soon as the icon popped up.

>I'm concerned that adding some kind of scanning icon onto it would make it hard to see other details (like missiles).

Agreed, and I wouldn't use an icon at all. Some sort of alteration in the ship's blip, be it coloring or pulsing.

>That being said, maybe we could put an indicator around the minimap so that there is an indicator that appears when a scan starts indicating what direction it is coming from?

I'm really just against giving players the directional info in the first place.

**Comment by Zitchas** [comment]
> That's fair, but the idea was actually to obfuscate the information a bit to retain some of the challenge. With the original idea, we get not only the exact direction of what's scanning you, but also the exact timing, which really doesn't make for a very interesting "mini-game" in my book at least.

This is a good point. The player doesn't currently get the timing, so let's leave that out.

How would giving them a direction / source of scan be in terms of gameplay if they don't get the timing?

(And to be fair, giving them the timing doesn't really make sense anyway now that I have thought a bit more about it. So let's scratch that of entirely)

**Comment by comnom** [comment]
>How would giving them a direction / source of scan be in terms of gameplay if they don't get the timing?

Definitely less hand holdy. From there there wouldn't be much need for any kind of obfuscation in the radar, and if only ships targeting the player "lit up", then you have the direction.

I'd still be iffy about an arrow or other icon pointing it out right away (because it's basically an arrow saying fly in the other direction), but maybe once you've targeted the source you could get some info in the targeting display.

It may come down to trying a few simple things and seeing how it plays out. I've got no real gripes with some extra info (especially since I agree that we shouldn't be doing much reading about moment to moment gameplay in the message text), just not all of it.

**Comment by Zitchas** [comment]
So long as the source of the scan is obvious on the radar, then I suppose a directional indicator wouldn't be necessary.

Although it'd be amusing to have a 2 ton "Advanced anti-scan triangulation unit" or something that added in such a directional indicator - at a price. Anyway, that's an amusing side idea, not a core thing for this.

How could we go about removing the text messages and replace them with a minimap indication?

**Comment by comnom** [comment]
>How could we go about removing the text messages and replace them with a minimap indication?

Here's a branch that removes the incoming messages and colors the radar: https://github.com/endless-sky/endless-sky/compare/master...comnom:radar-scan . You can try playing with that.

**Comment by damccull** [comment]
Hello,

New to the game, but old hat to EV and it's sequels.

I like the idea of some kind of directional indicator simply based on realism. However, it could be an upgrade requirement. Take airplanes, Mode-C transponders and radar in real life.

The airplane is constantly getting hit with radar sweeps and is completely oblivious to it. However, the Mode-C transponder can pick up a specific type of radar sweep (an interrogation) and reply to it. Even this is omni-directional. The game could mimic this behavior by default, where the ship must have a piece of equipment (a receiver) installed to even know it's being scanned in the first place.

Then, if you want to also know the direction of the scan, you buy the upgraded version of the equipment which includes multiple discreet sensors (lore) around the ship that will pick up the scan at different times, correlate the timing of each sensor's pickup, and triangulate where the scan should have come from. This can be done in several levels, of course:

* 1 sensor could provide an estimated distance based on known scanners in a particular frequency range (leaving room open for unidentified scanners to hide the user).
* 2 sensors would provide a quadrant around the ship.
* 3 sensors would provide a small arc in the direction of the scanner.

This could set up some additional gameplay where scanners themselves could have a resolution based on the quality of the scanner, and the time needed to scan is based on the resolution of the scanner and the distance to the ship being scanned. So a private militia in an independent system, for instance, might have only a low-resolution scanner and if over a half-screen away, might need more than 30 seconds to finish the scan, while being right next to the player would only require 3-5 seconds. The Syndicate, on the other hand, with access to major tech, could have a high resolution scanner that finishes the half-screen scan in 10 seconds, while finishing up close in only 1-2 seconds.

I assume most of this post would require major changes to the scanner system, but I wanted to suggest it since you all were looking at how to make scanner-dodging fun for the player.

In the end, give us some kind of directional indicator, even if we have to upgrade our sensors to get it. It's only realistic.

**Comment by comnom** [comment]
@damccull

Hey man. I'm not entirely opposed to an explicit directional indicator (although I am opposed to it being in the center of the main panel like the outgoing scan element), but neither am I convinced it's strictly necessary yet.

In my limited testing with the branch I linked above, it's pretty easy to tell the direction of an incoming scan but we'll need more data points from others to really know.

Right now though, however we cue the player to an incoming scan it's still nigh impossible to do any in-system dodging. The only evasive maneuver I found was to just jump out with a scram.

My intention was to more thoroughly test things; namely, using interference plating and lengthening scan times, but I got pulled in a few different directions with my vanilla work and now real life is in the way.

So long story short, I think a radar indication is enough for notifying the player, but making scanning fun and dodgeable will take more testing and adjustments outside of that, and that's more than I can do right now.

**Comment by Galaucus** [comment]
https://github.com/Endless-Sky-Pirate-Campaign/endless-sky/pull/5

Check this out, it's an overhaul of the whole scanning message.

**Comment by yafflehk** [comment]
It would be neat to make dodging a scan more part of the game, hiding near suns and planets to confuse scanners, scanner range decreases in asteroid rings, shutting down or shielding systems to run silent but with a performance penalty, maybe even disposable drones with a fake ship signature to distract scanners, all kinds of fun stuff.

**Comment by samrocketman** [comment]
A large part of the problem isn't necessarily the message clog.  It's the frequency at which fleets are scanned.  For example, hang out in Remnant space for a few minutes and you'll find your fleet being followed by 50 Petrels constantly scanning in a confusing frenzy that never ends.

Instead, scan limits should be put in place per system per government.  There's no need for the same faction to scan one player ship 100+ times.  It need only scan once to fill information in a government database.

### Proposal

I propose that once every ship in your fleet has been scanned once the government stops attempting to scan.  This would help manage the frenzy and ship log cluttering two fold.  It will still be annoying for mega fleets but that's probably fine.  Better than before.  50 Petrels will quickly scan through a mega fleet.

**Comment by Amazinite** [comment]
#8615 updated the scan messages so that you're no longer notified if one of your escorts is scanned and they don't have an illegal cargo/outfits for you to be concerned about.
#9714 greatly reduces the number of scans that can occur each day by limiting surveillance personality fleets to one scan per ship, just like non-surveillance personality fleets.
